### DISCUSSION TITLE
E: Lzma_read Read Error (7)  - package lists or status file could not be parsed or opened

### DISCUSSION THREAD

**Discussion by Vinalti** [rickysarraf/apt-offline#166] [Q&A]
## Summary
When installing updates with `apt-offline` on an old computer running `Kubuntu 18.04` that wasn't use and updated for a while, apt source files are getting broken, apt cannot parse them. It seems to be due to a change in the format used by apt-offline.

Using the classical process:
```sh
# On Offline PC
apt-offline set update.sig

# On Internet PC
apt-offline get --bundle updates.zip update.sig

# On Offline PC
apt-offline install updates.zip
```

## Problem
After  installation of the updates with apt-offline :

```sh
# On Offline PC
sudo apt-offline set --upgrade upgrade.zip
# Gathering details for the 'upgrade' operation
# E: Lzma_read Read Error (7) 
# E: The package lists or status file could not be parsed or opened.
# ERROR: FATAL: Something is wrong with the APT system
```
Almost all APT commands give the same result. (the two `E:` lines)

## Source of the Problem

It seems that the source format was changed.
```markdown
> 1.8.2-1
> Superseded in sid-release on 2020-04-30
# (...) skip to line 30
  * Switch to 3.0 (quilt) source format
  * Update debian/gbp.conf about switch to 3.0 (quilt) source format
```

I think that the apt sources files changed format but it seems to miss backward compatibility with the previous format.

## Recommended Solution
Would it be possible to add an option for selecting another source file format version, in order to assure backward compatibility with old versions of apt ?

**Comment by Vinalti** [comment]
### Analysis
Rollback to  version `1.8.1` (`apt-offline_1.8.1_all.deb`) fixes the issue, confirming that the version 1.8.2 introduced it.

**Comment by rickysarraf** [comment]
Thanks for the report though it is quite odd. I'll look when I'm back to my
machine. Meanwhile, request you to share the full outputs along with the
--verbose option

s3nt fr0m a $martph0ne, excuse typ0s

**Comment by Vinalti** [comment]
Unfortunately I cannot share the output. But I can tell you that the apt version was upgraded from version `1.6.12ubuntu0.1` to `1.6.14`  [changelog](https://launchpad.net/ubuntu/+source/apt/+changelog). After the upgrade of apt, I could use the last version of apt-offline and the generated source files were understood by the new apt.

I couldn't find a reference to  `quilt`, `3.0`  or `source format` in the commit logs, do you where this feature was implemented ?

**Comment by rickysarraf** [comment]
`apt-offline` will cycle through the known archive compressor types depending on what the remote server offers. Every apt repository server is independent to choose what compressor format they want to offer.

In your case, the remote server offered xz/lzma compressor, which as you mentioned, wasn't supported by apt in version `1.6.12`. And the later version of `apt` introduced that functionality upon which there was no issue.

There's not much `apt-offline` can do here. If we add an option to enforce a single compressor type, in this use case, it'd have failed with `404` errors. That's why we choose to cycle through all known supported compressor formats before giving up.


>I couldn't find a reference to quilt, 3.0 or source format in the commit logs, do you where this feature was implemented ?

This is a completely different thing, particularly related to Debian packaging only. It has nothing directly to do with `apt` or `apt-offline`.

**Comment by rickysarraf** [comment]
There's not much to do here. I'm converting this to a discussion instead.
